src/components/CheckBox.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { View, Text, Image, StyleSheet } from 'react-native';
+import { RectButton } from 'react-native-gesture-handler';
+
+import { icons, theme } from '../constants';
+
+const { FONTS, COLORS } = theme;
+const BOX_SIZE = 11;
+
+const CheckBox = ({ label }) => {
+    const [ isChecked, setIsChecked ] = useState(false);
+
+    return (
+        <View style={styles.checkBoxContainer}>
+            <RectButton 
+                style={[styles.box, 
+                    { 
+                        backgroundColor: isChecked 
+                        ? COLORS.primary
+                        : COLORS.white,
+                        borderColor: isChecked
+                        ? COLORS.primary
+                        : COLORS.black
+                    }]}
+                onPress={() => setIsChecked(!isChecked)}
+            >
+                { isChecked ? 
+                    <Image 
+                        source={icons.check} 
+                        style={styles.icon}
+                        resizeMode="contain"
+                    />
+                    : <View/>
+                }
+            </RectButton>
+            <View style={styles.labelContainer}>
+                <Text style={styles.textLabel}>{ label }</Text>
+            </View>
+        </View>
```

Comment:
The CodeClimate tells them that you can try refactoring here 😊 

![image](https://user-images.githubusercontent.com/69128578/114465094-4210a780-9bac-11eb-95be-1510f669a9f3.png)